### PR TITLE
#1063 Pareto_nbd distribution enhancements

### PR DESCRIPTION
**Pull request by Ishaanjolly**
Distribution new customer enhancements

## Description

I added `n_samples: int = 1000` to `_distribution_new_customers` and modified this line within: 
```
        if dataset.sizes["chain"] == 1 and dataset.sizes["draw"] == 1:
            # For map fit add a dummy draw dimension
            dataset = dataset.squeeze("draw").expand_dims(draw=range(n_samples))
```
Additionally:  
I added `n_samples: int = 1` to `distribution_new_customer_recency_frequency` and modified this: 

```
self._distribution_new_customers(
            data=data,
            T=T,
            random_seed=random_seed,
            var_names=["recency_frequency"],
            n_samples=n_samples,
        )["recency_frequency"]
```

## Related Issue

- [ ] Closes #
- [x] Related to #1063

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1067.org.readthedocs.build/en/1067/

### CODE REVIEW

**Review by Ishaanjolly — COMMENTED** [review]
had to do this again as not updated the branch!

**Comment by juanitorduz** [comment]
Thanks @Ishaanjolly ! 

We need to fix one test that is failing

```python
 tests/clv/models/test_pareto_nbd.py::TestParetoNBDModel::test_posterior_distributions[map] - assert (1, 1, 2357, 2) == (1, 1000, 2357, 2)
```

Let me know if you need help this that :)

**Comment by Ishaanjolly** [comment]
> Thanks @Ishaanjolly !
> 
> We need to fix one test that is failing
> 
> ```python
>  tests/clv/models/test_pareto_nbd.py::TestParetoNBDModel::test_posterior_distributions[map] - assert (1, 1, 2357, 2) == (1, 1000, 2357, 2)
> ```
> 
> Let me know if you need help this that :)


I had set n_samples = 1 in the `_distribution_new_customers_frequency method` without looking at the requirement.. think it's good now!

**Comment by ColtAllen** [comment]
Have you checked the Pareto/NBD notebook to make sure no changes need to be made?

**Comment by juanitorduz** [comment]
> Have you checked the Pareto/NBD notebook to make sure no changes need to be made?

The notebooks are running on the CI so everything looks good :)

File: pymc_marketing/clv/models/pareto_nbd.py [hunk 1]
```diff
@@ -1052,6 +1056,8 @@ def distribution_new_customer_recency_frequency(
             Not required if `data` Dataframe contains a `T` column.
         random_seed : ~numpy.random.RandomState, optional
             Random state to use for sampling.
+        n_samples : int, optional
+            Number of samples to generate. Defaults to 1.
```
Comment (juanitorduz) [inline pymc_marketing/clv/models/pareto_nbd.py:1060]: Defaults to 1_000 right?

**Review by juanitorduz — CHANGES_REQUESTED** [review]

File: pymc_marketing/clv/models/pareto_nbd.py (hunk 1 above)
Comment (Ishaanjolly) [reply pymc_marketing/clv/models/pareto_nbd.py:1060]: 1000, yeah sorry missed it. will correct it.

Comment (Ishaanjolly) [reply pymc_marketing/clv/models/pareto_nbd.py]: done!

**Review by juanitorduz — APPROVED** [review]

**Comment by juanitorduz** [comment]
Thanks @Ishaanjolly !

**Pull request merged**